### DISCUSSION TITLE
Bitwarden CLI Docker image

### DISCUSSION THREAD

**Discussion by PillFall** [bitwarden/clients#8344] [Password Manager]
### Code Contribution Proposal

I was wondering why there is no Docker image for the CLI. I am sure this is way useful for a lot of people.

For that reason, I just created a pretty basic Dockerfile to build the smallest docker image I have been able to get.

```Dockerfile
# Smallest glibc based image
# 4.25MB on Linux x86_64
#
# After the image is build
# 129MB on Linux x86_64
FROM busybox:glibc

# Assume the executables are in the current folder. Can be improved downloading from GitHub or simmilar.
COPY --chmod=755 bw /usr/local/bin/bw
COPY --chmod=755 bws /usr/local/bin/bws

# Import missing libs from ubuntu image. Assume x86_64 architecture.
COPY --from=ubuntu:latest /lib/x86_64-linux-gnu/libgcc_s.so.1 /lib/libgcc_s.so.1
COPY --from=ubuntu:latest /lib/x86_64-linux-gnu/libdl.so.2 /lib/libdl.so.2
COPY --from=ubuntu:latest /lib/x86_64-linux-gnu/libstdc++.so.6 /lib/libstdc++.so.6
```

Hope this helps to create an official Docker image for Bitwarden CLI.

**Comment by ahgraber** [comment]
+1, and noting prior community requests that may have been lost in the migration from bitwarden/cli to bitwarden/clients

https://community.bitwarden.com/t/include-docker-image-with-cli-client/40400/4
https://github.com/bitwarden/cli/issues/214

My use case: i'd like to use the bitwarden cli in association with external secrets for k8s (https://external-secrets.io/latest/examples/bitwarden/), and I'd much prefer to use a bitwarden-provided container.
